media-libs/opencv/opencv-4.8.1-r1.ebuild

```diff
@@ -0,0 +1,781 @@
+	${X86_CPU_FEATURES[@]}
+)
+IUSE="${IUSE} ${CPU_FEATURES_MAP[@]%:*}"
+
+# OpenGL needs gtk or Qt installed to activate, otherwise build system
+# will silently disable it without the user knowing, which defeats the
+# purpose of the opengl use flag.
+# cuda needs contrib, bug #701712
+REQUIRED_USE="
+	cpu_flags_x86_avx2? ( cpu_flags_x86_f16c )
+	cpu_flags_x86_f16c? ( cpu_flags_x86_avx )
+	cuda? (
+		contrib
+		tesseract? ( opencl )
+	)
+	cudnn? ( cuda )
+	dnnsamples? ( examples )
+	gflags? ( contrib )
+	glog? ( contrib )
+	contribcvv? ( contrib || ( qt5 qt6 ) )
+	contribdnn? ( contrib )
+	contribfreetype? ( contrib )
+	contribhdf? ( contrib )
+	contribovis? ( contrib )
+	contribsfm? ( contrib eigen gflags glog )
+	contribxfeatures2d? ( contrib )
+	java? ( python )
+	opengl? ( || ( qt5 qt6 ) )
+	python? ( ${PYTHON_REQUIRED_USE} )
+	tesseract? ( contrib )
+	?? ( gtk3 || ( qt5 qt6 ) )
+	?? ( cuda lto )
+	?? ( cuda openexr )
+	?? ( cuda tbb )
+"
+
+# The following logic is intrinsic in the build system, but we do not enforce
+# it on the useflags since this just blocks emerging pointlessly:
+#	openmp? ( !tbb )
+
+RDEPEND="
+	app-arch/bzip2[${MULTILIB_USEDEP}]
+	<dev-libs/protobuf-23:=[${MULTILIB_USEDEP}]
+	sys-libs/zlib[${MULTILIB_USEDEP}]
+	cuda? ( dev-util/nvidia-cuda-toolkit:0= )
+	cudnn? ( dev-libs/cudnn:= )
+	contribdnn? ( dev-libs/flatbuffers:= )
+	contribhdf? ( sci-libs/hdf5:= )
+	contribfreetype? (
+		media-libs/freetype:2[${MULTILIB_USEDEP}]
+		media-libs/harfbuzz:=[${MULTILIB_USEDEP}]
+	)
+	contribovis? ( >=dev-games/ogre-1.12:= )
+	ffmpeg? ( media-video/ffmpeg:0=[${MULTILIB_USEDEP}] )
+	gdal? ( sci-libs/gdal:= )
+	gflags? ( dev-cpp/gflags:=[${MULTILIB_USEDEP}] )
+	glog? ( dev-cpp/glog:=[${MULTILIB_USEDEP}] )
+	gphoto2? ( media-libs/libgphoto2:=[${MULTILIB_USEDEP}] )
+	gstreamer? (
+		media-libs/gstreamer:1.0[${MULTILIB_USEDEP}]
+		media-libs/gst-plugins-base:1.0[${MULTILIB_USEDEP}]
+	)
+	gtk3? (
+		dev-libs/glib:2[${MULTILIB_USEDEP}]
+		x11-libs/gtk+:3[${MULTILIB_USEDEP}]
+	)
+	ieee1394? (
+		media-libs/libdc1394:=[${MULTILIB_USEDEP}]
+		sys-libs/libraw1394[${MULTILIB_USEDEP}]
+	)
+	java? ( >=virtual/jre-1.8:* )
+	jpeg? ( media-libs/libjpeg-turbo:=[${MULTILIB_USEDEP}] )
+	jpeg2k? ( media-libs/openjpeg:2=[${MULTILIB_USEDEP}] )
+	lapack? (
+		|| (
+			(
+				virtual/cblas
+				>=virtual/lapack-3.10
+				virtual/lapacke
+			)
+			(
+				sci-libs/atlas
+			)
+		)
+	)
+	opencl? (
+		virtual/opencl[${MULTILIB_USEDEP}]
+		dev-util/opencl-headers
+	)
+	openexr? (
+		dev-libs/imath:=
+		media-libs/openexr:=
+	)
+	opengl? (
+		virtual/opengl[${MULTILIB_USEDEP}]
+		virtual/glu[${MULTILIB_USEDEP}]
+	)
+	png? ( media-libs/libpng:0=[${MULTILIB_USEDEP}] )
+	python? (
+		${PYTHON_DEPS}
+		dev-python/numpy[${PYTHON_USEDEP}]
+	)
+	qt5? (
+		dev-qt/qtgui:5=
+		dev-qt/qtwidgets:5=
+		dev-qt/qttest:5=
+		dev-qt/qtconcurrent:5=
+		opengl? ( dev-qt/qtopengl:5= )
+	)
+	!qt5? (
+		qt6? (
+			dev-qt/qtbase:6=[gui,widgets,concurrent,opengl?]
```

Comment:
Qt packages have subslots, but they also have metadata description:
```
        <slots>
                <subslots>
                        Must only be used by packages that are known to use private parts of the Qt API.
                </subslots>
        </slots>
```

This would be a good opportunity to establish that need. `grep -Ri include.*qt.*priv`, grep -Ri include.*qt.*_p and `grep -Ri include.*qpa` all returned nothing so you can drop all slot ops.